### PR TITLE
[Core/Spells]: Fix glyph of arcane shot not refunding mana.

### PR DESCRIPTION
**Pull request by lineagedr**
## Changes Proposed:
-  Add arcane shot spell script.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/6622
- closes https://github.com/chromiecraft/chromiecraft/issues/1020

## Tests Performed:

- Logged in on a hunter applied glyph of arcane shot then shot a dummy with my arcane shot and got 20% of the mana refunded.

## How to Test the Changes:

1. Login on a hunter
2. Apply glyph of arcane shot
3. Use arcane shot on a enemy target
4. Check if you've been refunded 20% of your mana

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Helias — APPROVED** [review]

**Review by UltraNix — APPROVED** [review]

File: data/sql/updates/pending_db_world/rev_1629141434760979400.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1629141434760979400');
+
+INSERT INTO `spell_proc_event` (`entry`, `SchoolMask`, `SpellFamilyName`, `SpellFamilyMask0`, `SpellFamilyMask1`, `SpellFamilyMask2`, `procFlags`, `procEx`, `ppmRate`, `CustomChance`, `Cooldown`) VALUES 
+(56841, 0, 9, 0x800, 0x800, 0x800, 0x100, 0, 0, 0, 0);
+
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) values (56841,'spell_hun_glyph_of_arcane_shot');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1629141434760979400.sql:6]: Always delete before insert.

File: src/server/scripts/Spells/spell_hunter.cpp [hunk 2]
```diff
@@ -1229,6 +1230,71 @@ class spell_hun_viper_attack_speed : public SpellScriptLoader
...
+public:
+    spell_hun_glyph_of_arcane_shot() : SpellScriptLoader("spell_hun_glyph_of_arcane_shot") {}
+
+    class spell_hun_glyph_of_arcane_shot_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_hun_glyph_of_arcane_shot_AuraScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo({ SPELL_HUNTER_GLYPH_OF_ARCANE_SHOT, SPELL_HUNTER_VICIOUS_VIPER });
```
Comment (Kitzunu) [inline src/server/scripts/Spells/spell_hunter.cpp:1245]: Why do you validate SPELL_HUNTER_VICIOUS_VIPER when you do not use it within the script?

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1629141434760979400.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1629141434760979400');
+
+DELETE FROM `spell_proc_event` WHERE `entry` = 56841;
+INSERT INTO `spell_proc_event` (`entry`, `SchoolMask`, `SpellFamilyName`, `SpellFamilyMask0`, `SpellFamilyMask1`, `SpellFamilyMask2`, `procFlags`, `procEx`, `ppmRate`, `CustomChance`, `Cooldown`) VALUES 
+(56841, 0, 9, 0x800, 0x800, 0x800, 0x100, 0, 0, 0, 0);
+
+DELETE FROM `spell_script_names` WHERE `spell_id` = 56841;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1629141434760979400.sql:7]:
```suggestion
DELETE FROM `spell_script_names` WHERE `spell_id` = 56841 AND `ScriptName` = 'spell_hun_glyph_of_arcane_shot';
```

**Review by Kitzunu — APPROVED** [review]

**Comment by lineagedr** [comment]
@Kitzunu good catch

**Review by Winfidonarleyan — APPROVED** [review]

**Pull request merged**